src/misc.cpp

```diff
@@ -144,17 +143,40 @@ class Logger {
 /// Version is empty.
 
 string engine_info(bool to_uci) {
-
-  const string months("Jan Feb Mar Apr May Jun Jul Aug Sep Oct Nov Dec");
-  string month, day, year;
-  stringstream ss, date(__DATE__); // From compiler, format is "Sep 21 2008"
-
-  ss << "Stockfish " << Version << setfill('0');
-
-  if (Version.empty())
+  /// For local dev compiles we try to append the commit sha and commit date
+  /// from git if that fails only the local compilation date is set and "nogit" is specified.
+  /// Stockfish YYYYMMDD-SHA
+  /// or
+  /// Stockfish YYYYMMDD-nogit
+  ///
+  /// dev CI builds will have the commit date + commit sha.
+  /// Stockfish YYYYMMDD-SHA
+  ///
+  /// For non dev builds include only the version number.
+  /// Stockfish version
+  stringstream ss;
+  ss << "Stockfish " << version << setfill('0') << "-";
```

Comment:
@vondele What you want is done here, though I just realised in case of a release version the `-` is superfluous, let me change that.